src/plugins/plugin.filler.js

```diff
@@ -232,47 +233,120 @@ function isDrawable(point) {
 	return point && !point.skip;
 }
 
-function drawArea(ctx, curve0, curve1, len0, len1) {
+function fillPointsSets(ctx, curve0, curve1, len0, len1, area, pointSets) {
 	var i, cx, cy, r;
+	var fillAreaPointsSet = [];
+	var clipAboveAreaPointsSet = [];
+	var clipBelowAreaPointsSet = [];
 
 	if (!len0 || !len1) {
 		return;
 	}
+	clipAboveAreaPointsSet.push({x: curve1[len1 - 1].x, y: area.top});
+	clipBelowAreaPointsSet.push({x: curve0[0].x, y: area.top});
+	clipBelowAreaPointsSet.push(curve0[0]);
 
 	// building first area curve (normal)
-	ctx.moveTo(curve0[0].x, curve0[0].y);
+	fillAreaPointsSet.push(curve0[0]);
 	for (i = 1; i < len0; ++i) {
-		helpers.canvas.lineTo(ctx, curve0[i - 1], curve0[i]);
+		curve0[i].flip = false;
+		fillAreaPointsSet.push(curve0[i]);
+		clipBelowAreaPointsSet.push(curve0[i]);
 	}
 
 	if (curve1[0].angle !== undefined) {
+		pointSets.fill.push(fillAreaPointsSet);
+		var radialSet = [];
```

Comment:
we typically define all variables at the top of a `function`. So even if you initialize here, we'd still declare `radialSet` along with `i, cx, cy, r`